File: backend/app/models/production.py
"""Production domain model — platform registry (independent of Teil-1 JSON scripts)."""

from __future__ import annotations

from datetime import datetime
from enum import StrEnum
from uuid import uuid4

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base


class ProductionStatus(StrEnum):
    DRAFT = "draft"
    ACTIVE_ELIGIBLE = "active_eligible"
    ARCHIVED = "archived"


class Production(Base):
    __tablename__ = "productions"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid4()))
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    slug: Mapped[str] = mapped_column(String(200), unique=True, nullable=False, index=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    status: Mapped[str] = mapped_column(
        String(32), nullable=False, default=ProductionStatus.DRAFT.value, index=True
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )
    archived_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

File: frontend/lib/types/production.ts
export type ProductionStatus = "draft" | "active_eligible" | "archived";

export type Production = {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  status: ProductionStatus;
  created_at: string;
  updated_at: string;
  archived_at: string | null;
};

export type ActiveProduction = {
  production_id: string | null;
  production: Production | null;
};

export type ProductionCreateInput = {
  name: string;
  slug?: string;
  description?: string;
};
